app/metrics/temperature_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 qbee.io
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics
+
+import (
+	"testing"
+)
+
+func TestCollectTemperature(t *testing.T) {
+
+	files, _ := getHwMonFiles()
+	if len(files) == 0 {
+		files, _ = getThermalZoneFiles()
+		if len(files) == 0 {
+			// No temperature files found, skip test. CI uses vms and they don't have temperature files.
+			t.Skip("no temperature files found")
+		}
+	}
+
+	metrics, err := CollectTemperature()
+
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
```

Comment:
It fails on my machine with:
```
=== RUN   TestCollectTemperature
    temperature_test.go:37: unexpected error: no temperature metrics found: no thermal zone metrics found
--- FAIL: TestCollectTemperature (0.00s)
```

I think the temperature metric collections should only produce an error when something unexpected happen (file exists but can't be opened due to permissions, etc...). But lack of thermal zones shouldn't cause an error, as not every device will have it.

Reply:
What I meant by my comment, was that the `CollectTemperature()` function shouldn't produce an error when there are no thermal zones, as lack of thermal zones can be normal for some devices. We don't want an error message in the agent logs when this happens.

Reply:
Skipping test on error

Reply:
@jonhenrik13 I'm not sure if you noticed my recent comment. We shouldn't produce an error when no temperature metrics are available.

Reply:
Not returning error, log in debug level